src/crate/crash/command.py

```diff
@@ -310,13 +316,22 @@ def _do_connect(self):
                                   cert_file=self.cert_file,
                                   key_file=self.key_file,
                                   ca_cert=self.ca_cert_file,
-                                  username=self.username)
+                                  username=self.username,
+                                  password=self.password)
         self.cursor = self.connection.cursor()
 
     def _connect(self, server):
         """ connect to the given server, e.g.: \connect localhost:4200 """
         self.last_connected_servers = server
-        self._do_connect()
+        try:
+            self._do_connect()
+        except (ProgrammingError, LocationParseError) as e:
+            if ('401 Client Error: Unauthorized' in e.args[0]):
+                self.username = input('Username: ')
+                self.password = getpass()
```

Comment:
as discussed: From a separation point of view it would be nice if the `CrateCmd` didn't do any user-input handling. That should be up to the `repl` (or `main` to establish the initial connection)